client/platform/desktop/frontend/components/MultiTrainingMenu.vue

```diff
@@ -2,15 +2,36 @@
 import type { DataTableHeader } from 'vuetify';
 
 import {
-  computed, defineComponent, onBeforeMount, set, del, reactive,
+  computed, defineComponent, onBeforeMount, set, del, reactive, ref,
 } from '@vue/composition-api';
-import { DatasetMeta, TrainingConfigs, useApi } from 'viame-web-common/apispec';
+import {
+  DatasetMeta, Pipelines, TrainingConfigs, useApi,
+} from 'viame-web-common/apispec';
+
 import { datasets } from '../store/dataset';
 
 export default defineComponent({
   setup(_, { root }) {
     const { getTrainingConfigurations, runTraining } = useApi();
 
+    const { getPipelineList } = useApi();
+    const unsortedPipelines = ref({} as Pipelines);
+    onBeforeMount(async () => {
+      unsortedPipelines.value = await getPipelineList();
+    });
+
+    const trainedPipelines = computed(() => {
+      const list: string[] = [];
+      Object.entries(unsortedPipelines.value).forEach(([, category]) => {
+        category.pipes.forEach((pipe) => {
+          if (pipe.type === 'trained') {
+            list.push(pipe.name);
+          }
+        });
+      });
+      return list;
+    });
```

Comment:
Needs to be slightly changed because I'm using an `includes` on an array and trained is an Category which contains an array of Pipes.  So it's either modify the `includes` to be a find or something else or modify the source.  I've adjusted to simplify the computed property with TS guaranteeing that if `trained` exists then `pipes` is an array..
```
interface Pipe {
  name: string;
  pipe: string;
  type: string;
}

interface Category {
  description: string;
  pipes: Pipe[];
}
```